src/Strategy/Maven.hs

```diff
@@ -177,12 +177,24 @@ getStaticAnalysis submoduleTargets closure = do
   filteredGraph <- applyMavenFilters submoduleTargets allSubmodules graph
   pure (filteredGraph, graphBreadth)
 
-applyMavenFilters :: Has (Reader MavenScopeFilters) sig m => Set Text -> Set Text -> Graphing MavenDependency -> m (Graphing Dependency)
+applyMavenFilters ::
+  ( Has Diagnostics sig m
```

Comment:
Even though the type is long I think it still looks better on one line. Did the formatter do this? It doesn't when I add the `Diagnostics` constraint. Maybe we can pair on seeing what's up if it was the formatter.

Reply:
The formatter didn't do this, I just thought that the convention  that we had in our code base , from what I've seen, was to have this portion as multi line when we have more than one constraint.

Reply:
I think it's somewhat a matter of taste, I'm not aware of a hard rule around more than one constraint. You should keep this if you think it's easier to read, that's more important than following a rule.

Reply:
I'm okay with having this as a one liner!